runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsPlugin.java

```diff
@@ -94,58 +94,60 @@ protected void shutDown() throws Exception
 	}
 
 	@Subscribe
-	public void onMenuEntryAdded(MenuEntryAdded menuEntryAdded)
+	public void onClientTick(ClientTick clientTick)
 	{
-		int type = menuEntryAdded.getType();
+		MenuEntry[] menuEntries = client.getMenuEntries();
 
-		if (type >= 2000)
+		for (MenuEntry entry : menuEntries)
 		{
-			type -= 2000;
-		}
-
-		int identifier = menuEntryAdded.getIdentifier();
-		if (type == SPELL_CAST_ON_PLAYER.getId()
-			|| type == ITEM_USE_ON_PLAYER.getId()
-			|| type == PLAYER_FIRST_OPTION.getId()
-			|| type == PLAYER_SECOND_OPTION.getId()
-			|| type == PLAYER_THIRD_OPTION.getId()
-			|| type == PLAYER_FOURTH_OPTION.getId()
-			|| type == PLAYER_FIFTH_OPTION.getId()
-			|| type == PLAYER_SIXTH_OPTION.getId()
-			|| type == PLAYER_SEVENTH_OPTION.getId()
-			|| type == PLAYER_EIGTH_OPTION.getId()
-			|| type == RUNELITE.getId())
-		{
-			Player[] players = client.getCachedPlayers();
-			Player player = null;
+			int type = entry.getType();
 
-			if (identifier >= 0 && identifier < players.length)
+			if (type >= 2000)
 			{
-				player = players[identifier];
+				type -= 2000;
 			}
 
-			if (player == null)
+			int identifier = entry.getIdentifier();
+			if (type == SPELL_CAST_ON_PLAYER.getId()
+				|| type == ITEM_USE_ON_PLAYER.getId()
+				|| type == PLAYER_FIRST_OPTION.getId()
+				|| type == PLAYER_SECOND_OPTION.getId()
+				|| type == PLAYER_THIRD_OPTION.getId()
+				|| type == PLAYER_FOURTH_OPTION.getId()
+				|| type == PLAYER_FIFTH_OPTION.getId()
+				|| type == PLAYER_SIXTH_OPTION.getId()
+				|| type == PLAYER_SEVENTH_OPTION.getId()
+				|| type == PLAYER_EIGTH_OPTION.getId()
+				|| type == RUNELITE.getId())
 			{
-				return;
-			}
+				Player[] players = client.getCachedPlayers();
+				Player player = null;
 
-			Decorations decorations = getDecorations(player);
+				if (identifier >= 0 && identifier < players.length)
+				{
+					player = players[identifier];
+				}
 
-			if (decorations == null)
-			{
-				return;
-			}
+				if (player == null)
+				{
+					continue;
+				}
 
-			MenuEntry[] menuEntries = client.getMenuEntries();
-			MenuEntry entry = menuEntries[menuEntries.length - 1];
+				Decorations decorations = getDecorations(player);
 
-			String oldTarget = entry.getTarget();
-			String newTarget = decorateTarget(oldTarget, decorations);
+				if (decorations == null)
+				{
+					continue;
+				}
 
-			entry.setTarget(newTarget);
+				String oldTarget = entry.getTarget();
+				String newTarget = decorateTarget(oldTarget, decorations);
 
-			client.setMenuEntries(menuEntries);
+				entry.setTarget(newTarget);
+			}
 		}
+
+		client.setMenuEntries(menuEntries);
```

Comment:
Since this is relatively expensive it would be better to only call `setMenuEntries` if you know it has changed.